### ISSUE TITLE
Fix uploading to App Store Connect

### ISSUE THREAD

**Issue by nickmeinhold** [nickmeinhold/crowdleague_public#10]
I order to solve the freeze described in #27 I am creating a new unlocked keychain as described in [here](https://stackoverflow.com/questions/58118395/github-action-macos-keychain-access) but now there are issues deploying.

## fastlane deliver 

The lane errors with "wrong username and password combination".  Try: 
- [x] don't set the new keychain as the default (no difference) 
- [ ] add the apple id password to the new keychain 
- [ ] instead of creating a new keychain, just unlock the default keychain 
- [x] set the FASTLANE_PASSWORD env var for the step 
  - this works!  doesn't make sense to me as the env var is set for the build and presumably works in the previous step where it is needed by Match 

## altool 

```sh
xcrun altool --upload-app --file "./ios/Runner.ipa" --username "cicd@crowdleague.app" --password "  ***" --verbose
```

```sh
2020-04-17 07:58:05.612 altool[19340:53090] Web service call (generateAppleConnectToken) result: {
    EnableJWTForAllCalls = 0;
    ErrorCode = "-20101";
    ErrorMessage = "Your account information was entered incorrectly. (-20101)";
    Errors =     (
        "Your account information was entered incorrectly. (-20101)"
    );
    RestartClient = 0;
    Success = 0;
}
```

**Comment by nickmeinhold** [comment]
Changing fastlane delivery Transporter settings doesn't appear to have worked. 

Operation timed out, see [Merge pull request #11 from nickmeinhold/i10 · nickmeinhold/crowdleague_public@8e17d70](https://github.com/nickmeinhold/crowdleague_public/runs/571010982?check_suite_focus=true)

**Comment by nickmeinhold** [comment]
Not sure if this is fixed, or was even a problem - it seems it's the build that freezes and I've been assuming it was the upload.

**Comment by nickmeinhold** [comment]
Moved the below out of the main comment, as the timeout was actually due to #27 

I assumed uploading to App StoreConnect was timing out but it was actually the build that was timing out due to the keychain being locked and asking for a password. I had tried: 
- [x] changing fastlane delivery Transporter settings as described in [deliver > firewall-issues](https://docs.fastlane.tools/actions/deliver/#firewall-issues) (also more info in
[upload_to_testflight > firewall-issues](https://docs.fastlane.tools/actions/upload_to_testflight/#firewall-issues)) 
- [x] changing how we force WebDAV publishing, by using an env var 
- [x] using altool instead (works locally, but fails in a workflow) 
- [ ] try other transporter options, see [Allow iTMSTransporter Transport parameters to be set · Issue #2445 · fastlane/fastlane](https://github.com/fastlane/fastlane/issues/2445) 
  - Note: [Issue #15375 · fastlane](https://github.com/fastlane/fastlane/issues/15375) seems to suggest that recent versions of fastlane have fixed any issues and there is no need for `-t DAV`
- [ ] don't use fastlane for upload and instead use the transporter directly

**Comment by nickmeinhold** [comment]
The initial problem was actually in the build (as discussed above).  The next problem was due to the env var not being set correctly.  

So I set the FASTLANE_PASSWORD env var for the step (as well as being already set for the build) and the value seems to then be available.  This doesn't make sense to me as the env var is set for the build and presumably works in the previous step where it is needed by Match.  Probably worth setting env vars per step from now on.
